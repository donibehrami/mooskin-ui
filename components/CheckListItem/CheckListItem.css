@import url('https://fonts.googleapis.com/icon?family=Material+Icons');

.checkitem {
    background: #f7f7f7;
    border-radius: 5px;
    padding: 20px;
    position: relative;
    margin-top: 5px;
}

.content {
    color: #aaa;
    font-size: 1rem;
    display: inline-block;
    vertical-align: middle;
    border-left: 1px solid #eaeaea;
    padding: 5px 0 2px 0;
    padding-left: 15px;
    margin-left: 15px;
    height: 25px;
}

.iconStyle {
    color: #dddddd;
    vertical-align: middle;
    font-size: 1.875rem;
    font-weight: 800;
}

.button {
    width: 40px;
    height: 40px;
    display: block;
    position: absolute;
    top: 12px;
    right: 12px;
    border-radius: 8px;
}