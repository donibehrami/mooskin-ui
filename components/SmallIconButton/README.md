# MooSkin UI - Small Icon Button Component

The MooSkin Small Icon Button Component works similar to the normal HTML `<button/>` element, but with pre-defined styling and attributes.

___

### Usage

To start using the Small Icon Button Component first you have to Import it

```
Import {SmallIconButton} from 'mooskin';
```
or
```
import SmallIconButton from 'mooskin/lib/SmallIconButton';
```

And then you can simply start using it by typing

```
<SmallIconButton attribute1="atr" attribute2="asd" />
```

For easy use, the components are named similar to normal HTML components but with a capital first letter.

Like the `<button/>` element it will accept given attributes and render differently based on the given attributes

### Examples


Making it disabled with a value, in this case the proper disabled class will be loaded (ex. disabling cursor)

```
<SmallIconButton value="This one is disabled" disabled />
```

or pass it a function for event handling

```
<SmallIconButton onClick={yourFunc} />
```

or just give it a custom style

```
<SmallIconButton style={yourStyle} />
```

but the most important one is to give it an icon

```
<SmallIconButton icon={'search'} />
```

give it an inverse style where only the icon is visible

```
<SmallIconButton inverseStyle />
```

## Supported attributes

* `id` - id of the element
* `icon` - icon for the button
* `disabled` - button should be disabled
* `className` - css class
* `inverseStyle` - a style variaton of the button 
* `style` - button style
* `onClick` - callback to be triggered on button click

Allthough these attributes are supported, all of them are optional.

## Supported icons (so far)

* `search`
* `add`
* `add box`
* `add circle`
* `add outline`
* `remove`
* `remove circle`
* `remove outline`
* `close`
* `star`
* `account`
* `build`
* `alarm`
* `alarm add`
* `alarm off`
* `alarm on`
* `done`
* `favorite`
* `home`
* `power`
* `settings`
* `verified`
* `visibility`
* `visibility off`
* `alert`
* `loop`
* `play`
* `pause`
* `stop`
* `contacts`
* `email`
* `mail outline`
* `forum`
* `message`
* `create`
* `block`
* `send`
* `download`
* `upload`
* `down`
* `up`
* `left`
* `right`
* `edit`
* `dropdown`
* `dropup`
* `check`
* `refresh`
* `sync`
* `group`
* `group add`
* `person`
* `person add`
* `notifications`
* `notifications active`
* `notifications none`
* `notifications off`
* `notifications paused`
* `check box`
* `indeterminate check box`
* `check box off`
* `radio checked`
* `radio unchecked`

#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)