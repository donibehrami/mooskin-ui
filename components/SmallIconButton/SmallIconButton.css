@import url('https://fonts.googleapis.com/icon?family=Material+Icons');
@import '../_utils/globals/variables.css';
@import '../_utils/globals/screens.css';
 
.button {
    font-family: 'Material Icons';
    cursor: pointer;
    display: inline-block;
    outline: 0;
    transition: background-color 400ms ease;
    vertical-align: middle;
}

.disabledButton.normalButton{
    cursor:not-allowed;
    background-color: #87dfed;
}

.disabledButton.transparent{
    cursor:not-allowed;
    color: #ccc;
}

.normalButton{
    background-color: var(--moosendBlue);
    border-radius: 50%;
    border: 1px solid transparent;
    color: #ffffff;
    margin: 2px;
    height: 28px;
    width: 28px;
    padding: 3px 2px;
}

.transparent{
    background-color: hsla(0, 100%, 50%, 0);;
    color: #bbb;
    font-size: 23px;
    border: 0;
    width: 23px;
    height: 23px;
    padding: 0px;
    margin: 3px;
}

.normalButton:hover:enabled {
    background-color: #4cb3c4;
}

.transparent:hover:enabled {
   color: var(--moosendBlue);
}

.icon {
    font-size: 21px;
}

@media (--small){
    .normalButton{
        height: 6vw;
        width: 6vw;
        padding: 0px;
    }

    .icon {
        font-size: 4.5vw;
    }

    .transparent{
        width: 6vw;
        height: 6vw;
    }
}

@media (--xSmall){
    .normalButton{
        height: 6vw;
        width: 6vw;
        padding: 0px;
    }

    .icon {
        font-size: 4.5vw;
    }

    .transparent{
        width: 6vw;
        height: 6vw;
    }
}