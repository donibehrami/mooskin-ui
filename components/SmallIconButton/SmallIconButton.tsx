import * as React from 'react';

import styles from './SmallIconButton.css';

export interface ISmallIconButtonProps {
    /** provide to make the button disabled */
    disabled?: boolean;

    /** provide to inverse the button's styles */
    transparent?: boolean;

    /** button icon */
    icon?: string;

    /** button id attribute */
    id?: string;

    /** button class */
    className?: string;

    /** override button styles */
    style?: {[key: string]: string};

    /** callback that is called when the button is clicked */
    onClick?: (e?: React.MouseEvent<HTMLElement>) => void;
}

export default class SmallIconButton extends React.Component<ISmallIconButtonProps, {}> {

    public static defaultProps = {
        className: '',
        style: {},
    };

    public render(){

        const {style, transparent, disabled, className, id, icon} = this.props;

        const buttonStyles = transparent ? styles.transparent : styles.normalButton;
        const disabledStyles = disabled ? styles.disabledButton : '';
        const iconStyle = icon ? `material-icons ${styles.icon}` : '';
        const iconFont = this.getIcon(icon);

        const classes = `button-icon-component
                         ${styles.button}
                         ${buttonStyles}
                         ${disabledStyles}
                         ${className}`;

        return (
            <button
                id={id}
                onClick={this.onClick}
                disabled={disabled}
                className={classes}
                style={style}
            >
                <i className={iconStyle} >{iconFont}</i>
            </button>
        );
    }

    private onClick = (e: React.MouseEvent<HTMLElement>) => {
        !this.props.disabled && this.props.onClick && this.props.onClick(e);
    }

    private getIcon = (icon?: string) => {
        return this.props.icon ? this.props.icon.replace(/\s/g, '_') : '';
    }
}
