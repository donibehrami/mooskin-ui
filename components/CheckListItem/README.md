# MooSkin UI - Check List Item Component

___

### Usage

To start using the Check List Item Component first you have to Import it

```
Import {CheckListItem} from 'mooskin';
```
or modular import
```
// JS
import CheckListItem from 'mooskin/lib/CheckListItem';

// CSS
import 'mooskin/lib/CheckListItem/style.css';
```

And then you can simply start using it by typing

```
<CheckListItem attribute1="atr" attribute2="asd" />
```

The Component will accept given attributes and render differently based on the given attributes

### Examples


Making it disabled, in this case the proper disabled class will be loaded (ex. disabling cursor)

```
<CheckListItem title="Do this" text="this has to be done" />
```

or pass it a function for event handling

```
<CheckListItem onClick={yourFunc} />
```

or just give it a custom style

```
<CheckListItem style={yourStyle} />
```

### Callback

The CheckListItem Component Callback will always return the boolean value of the CheckListItem when it changes. This can be used with a function passed via the `onClick` prop. For example, if u want to fire a function if the Switch is turned on (oh yea), pass a function to the `onClick` prop.

```
const onCheck = (e, data) => {      // data is the callback object, which consists of value and a dataLabel(not required)
    if(data.value){                 // this can be either true or false, so if the CheckListItem checked(done) or not
        // do something
    }      
};

<CheckListItem onClick={onCheck} />
```
In this case the statement block will get fired if the CheckListItem is checked(done).

This can be used in various situations and combinations, for an enhanced development experience.

<div class="playground-doc">

## Supported attributes 

* `id` - id of the element
* `className` - css class
* `status` - status of the check list item (active/nonactive, done/undone)
* `title` - Check list item title
* `text` - Check list item text
* `style` - button style
* `onClick` - callback to be triggered on button click

</div>

Allthough these attributes are supported, all of them are optional.

#### For more

___

[MooSkin-UI](https://github.com/moosend/mooskin-ui)