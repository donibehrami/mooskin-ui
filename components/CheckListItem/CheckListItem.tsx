import * as React from 'react';

import {H2} from '../Headings/index';
import SmallIconButton from '../SmallIconButton/SmallIconButton';
import styles from './CheckListItem.css';

export interface ICheckListItemProps {

    /** check list item id attribute */
    id?: string;

    /** item status */
    status?: boolean;

    /** item title */
    title?: string;

    /** item text */
    text?: string;

    /** check list item class */
    className?: string;

    /** override check list item styles */
    style?: {[key: string]: string};

    /** callback that is called when the button is clicked */
    onClick?: (e?: React.MouseEvent<HTMLInputElement>) => void;

}

export default class CheckListItem extends React.Component<ICheckListItemProps, {}> {

    public static defaultProps = {
        className: '',
        status: false,
        style: {}
    };

    public render(){

        const {status, id, text, title, className, style} = this.props;
        const {checkitem, iconStyle, content, button} = styles;
        const icon = status ? 'check' : 'close';

        return (
            <div
                className={className}
                style={style}
            >
                <H2 style={{marginLeft: '5px'}}>{title}</H2>
                <div className={checkitem} id={id}>
                    <i className={`material-icons ${iconStyle}`} >{icon}</i>
                    <div className={content}>
                        <span>{text}</span>
                        <SmallIconButton
                            className={button}
                            icon="create"
                            onClick={this.onClick}
                        />
                    </div>
                </div>
            </div>
        );
    }

    private onClick = (e: React.MouseEvent<HTMLInputElement>) => {
        this.props.onClick && this.props.onClick(e);
    }

}
