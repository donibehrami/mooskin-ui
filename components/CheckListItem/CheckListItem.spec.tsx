import * as React from 'react';
import SmallIconButton from '../SmallIconButton';
import CheckListItem from './CheckListItem';

import { shallow } from 'enzyme';

describe('CheckListItem', () => {

    test('renders into dom with title text and status', () => {

        const component = shallow(<CheckListItem title="One Ring" text="Toss the One Ring into Mt. Doom" />);

        expect(component.find('div')).toHaveLength(3);
        expect(component.find('span').text()).toBe('Toss the One Ring into Mt. Doom');
        expect(component.find('i').hasClass('fa fa-times fa-lg')).toBe(true);
    });

    test('callback function called when check item button is clicked', () => {
        const func = jest.fn();

        const component = shallow(<CheckListItem onClick={func} className={'myClass'} status/>);

        expect(component.find('i').hasClass('fa fa-check fa-lg')).toBe(true);
        component.find(SmallIconButton).simulate('click');
        expect(func).toHaveBeenCalled();
    });

});
