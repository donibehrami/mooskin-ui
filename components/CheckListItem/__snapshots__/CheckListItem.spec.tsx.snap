// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckListItem renders correctly 1`] = `
<div
  className="checkitem-component myClass"
  style={
    Object {
      "color": "blue",
    }
  }
>
  <H2
    className=""
    style={
      Object {
        "display": "",
        "marginLeft": "5px",
      }
    }
  >
    check item title
  </H2>
  <div
    className="checkitem"
    id="check1"
  >
    <i
      className="material-icons iconStyle"
    >
      check
    </i>
    <div
      className="content"
    >
      <span>
        check item text
      </span>
      <SmallIconButton
        className="button"
        icon="create"
        onClick={[Function]}
        style={Object {}}
      />
    </div>
  </div>
</div>
`;
